ports/lua/CMakeLists.txt

```
@@ -1,117 +1,256 @@
-            TARGET_COMPILE_DEFINITIONS (lua PUBLIC -DLUA_USE_DLOPEN)
-            TARGET_LINK_LIBRARIES (lua ${CMAKE_DL_LIBS})
-        ENDIF ()
-        IF (LIB_MATH)
-            TARGET_LINK_LIBRARIES (lua m)
-        ENDIF ()
-    ENDIF ()
-ENDIF ()
-
-INSTALL ( TARGETS lua
-    EXPORT unofficial-lua-config
-    RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
+
+cmake_minimum_required(VERSION 3.28)
+
+project(
+    lua-c-library
+    LANGUAGES C
 )
 
-install(EXPORT unofficial-lua-config DESTINATION share/unofficial-lua)
+set(CMAKE_C_STANDARD 99)
 
-# CXX library
-IF (COMPILE_AS_CPP)
-    ADD_SUBDIRECTORY(cpp)
-ENDIF()
+set(LUA_SOURCE_DIR "${CMAKE_SOURCE_DIR}/src/")
+
+set(HDR_LIBLUA
+    "${LUA_SOURCE_DIR}/lapi.h"
+    "${LUA_SOURCE_DIR}/lauxlib.h"
+    "${LUA_SOURCE_DIR}/lcode.h"
+    "${LUA_SOURCE_DIR}/lctype.h"
+    "${LUA_SOURCE_DIR}/ldebug.h"
+    "${LUA_SOURCE_DIR}/ldo.h"
+    "${LUA_SOURCE_DIR}/lfunc.h"
+    "${LUA_SOURCE_DIR}/lgc.h"
+    "${LUA_SOURCE_DIR}/llex.h"
+    "${LUA_SOURCE_DIR}/llimits.h"
+    "${LUA_SOURCE_DIR}/lmem.h"
+    "${LUA_SOURCE_DIR}/lobject.h"
+    "${LUA_SOURCE_DIR}/lopcodes.h"
+    "${LUA_SOURCE_DIR}/lparser.h"
+    "${LUA_SOURCE_DIR}/lstate.h"
+    "${LUA_SOURCE_DIR}/lstring.h"
+    "${LUA_SOURCE_DIR}/ltable.h"
+    "${LUA_SOURCE_DIR}/ltm.h"
+    "${LUA_SOURCE_DIR}/lua.h"
+    "${LUA_SOURCE_DIR}/luaconf.h"
+    "${LUA_SOURCE_DIR}/lualib.h"
+    "${LUA_SOURCE_DIR}/lundump.h"
+    "${LUA_SOURCE_DIR}/lvm.h"
+    "${LUA_SOURCE_DIR}/lzio.h"
+)
+
+set(PUBLIC_HDR_LIBLUA
+    "${LUA_SOURCE_DIR}/lualib.h"
+    "${LUA_SOURCE_DIR}/lua.h"
+    "${LUA_SOURCE_DIR}/luaconf.h"
+    "${LUA_SOURCE_DIR}/lauxlib.h"
+    "${LUA_SOURCE_DIR}/lua.hpp"
+)
+
+set(SRC_LIBLUA
+    "${LUA_SOURCE_DIR}/lapi.c"
+    "${LUA_SOURCE_DIR}/lauxlib.c"
+    "${LUA_SOURCE_DIR}/lbaselib.c"
+    "${LUA_SOURCE_DIR}/lbitlib.c"
+    "${LUA_SOURCE_DIR}/lcode.c"
+    "${LUA_SOURCE_DIR}/lcorolib.c"
+    "${LUA_SOURCE_DIR}/lctype.c"
+    "${LUA_SOURCE_DIR}/ldblib.c"
+    "${LUA_SOURCE_DIR}/ldebug.c"
+    "${LUA_SOURCE_DIR}/ldo.c"
+    "${LUA_SOURCE_DIR}/ldump.c"
+    "${LUA_SOURCE_DIR}/lfunc.c"
+    "${LUA_SOURCE_DIR}/lgc.c"
+    "${LUA_SOURCE_DIR}/linit.c"
+    "${LUA_SOURCE_DIR}/liolib.c"
+    "${LUA_SOURCE_DIR}/llex.c"
+    "${LUA_SOURCE_DIR}/lmathlib.c"
+    "${LUA_SOURCE_DIR}/lmem.c"
+    "${LUA_SOURCE_DIR}/loadlib.c"
+    "${LUA_SOURCE_DIR}/lobject.c"
+    "${LUA_SOURCE_DIR}/lopcodes.c"
+    "${LUA_SOURCE_DIR}/loslib.c"
+    "${LUA_SOURCE_DIR}/lparser.c"
+    "${LUA_SOURCE_DIR}/lstate.c"
+    "${LUA_SOURCE_DIR}/lstring.c"
+    "${LUA_SOURCE_DIR}/lstrlib.c"
+    "${LUA_SOURCE_DIR}/ltable.c"
+    "${LUA_SOURCE_DIR}/ltablib.c"
+    "${LUA_SOURCE_DIR}/ltm.c"
+    "${LUA_SOURCE_DIR}/lundump.c"
+    "${LUA_SOURCE_DIR}/lutf8lib.c"
+    "${LUA_SOURCE_DIR}/lvm.c"
+    "${LUA_SOURCE_DIR}/lzio.c"
+)
+
+set_source_files_properties(${SRC_LIBLUA} PROPERTIES LANGUAGE C)
+
+add_library(
+    lua
+    ${SRC_LIBLUA}
+)
```

Nitpick: Keep short statements short.
```suggestion
add_library(lua ${SRC_LIBLUA})
```

---

Made short statements shorter.